broke-broker/src/main/java/org/dsngroup/broke/broker/Server.java

```diff
@@ -110,8 +110,6 @@ public void close() {}
 
```

Comment:
Take a look at here, the `workerGroup` and `bossGroup` of netty are gracefully shutdown. In the meaning, do we still need to explicit close out channels?

Reply:
In other word, take it a try that if server shutdown, the connection of client will be close or not?

Reply:
Currently not. I still can't figure out how the client can trigger the ```closeFuture()``` listener. The client will only close when the **IO operations** on the channel failed.

Reply:
It's because the server side don't send the close out signal.

Reply:
Netty cannot detect an unexpected channel close. I think the client can detect a server close by using ```PINGREQ```.  For example, if the client hasn't received ```PINGREQ``` for a limited time, close itself.

Reply:
I found that the client can override ```channelInactive()``` at inbound handler for handling the server shutdown.

Reply:
Ok, but it comes with the scope of ttl mechanism, I think you may not necessarily implement this. Just ensure the gracefully shut-down will clean up resources in server.